lib/nationbuilder/method.rb

```diff
@@ -40,7 +40,13 @@ def method_args(args)
   def nonmethod_args(args)
     a = {}
     args.each do |k, v|
+      if k == :custom_values
+        v.each do |custom_v|
```

Comment:
Are we passed a Hash here? Ruby will yield the key and value if you give two params, e.g.

```Ruby
   v.each do |custom_k, custom_v|
     memo[:"custom_values[#{custom_k.to_s}]"] = custom_v
   end
```

It's all the same of course, but explicitly naming the key and the value is less surprising to the reader than array indexes.

Reply:
Hey Sean, thanks for commenting! Good call, v is a hash like `{custom_field_slug: "custom_field_value"}` or potentially multiple, like `{custom_field_slug: "custom_field_value", custom_field_slug2: "custom_field_value2"}` so this can definitely be cleaned up like you suggested. I'll resubmit! Cheers.

Reply:
👋 catching up a bit/looking at this some more - so the issue seems to be the query params not being encoded properly (ie. the nested hash encoding as a hash, not as the server expects). So what's needed is effectively something along the lines of `ActiveSupport` [`to_query`](https://apidock.com/rails/Hash/to_query) or `Rack`'s [`build_nested_query`](http://www.rubydoc.info/gems/rack/Rack/Utils#build_nested_query-class_method).

Without adding additional dependencies, this could be done in Ruby following the design in `Rack` or akin to [how it's done here](https://mensfeld.pl/2012/01/converting-nested-hash-into-http-url-params-hash-version-in-ruby/) (perhaps as a `Utils` module), though it feels like that could/should be applied within client when [we append the args to the query](https://github.com/nationbuilder/nationbuilder-rb/blob/master/lib/nationbuilder/client.rb#L57) instead of handling it as a special case in Method - thoughts?